hip_data_tools/apache/cassandra.py

```diff
@@ -32,6 +38,97 @@ def get_ssl_context(cert_path: str) -> SSLContext:
     return ssl_context
 
 
+_PYTHON_TO_CASSANDRA_DATA_TYPE_MAP = {
+    "Timestamp": "timestamp",
+    "str": "varchar",
+    "int64": "bigint",
+    "int32": "int",
+    "dict": "map",
+    "float64": "double",
+    "UUID": "UUID",
+}
+
+
+def _get_data_frame_column_types(data_frame):
+    data_frame_col_dict = {}
+    for col in data_frame:
+        data_frame_col_dict[col] = type(data_frame[col][0]).__name__
+    return data_frame_col_dict
+
+
+def convert_dataframe_columns_to_cassandra(data_frame):
+    """
+    Extracts a dictionary of column names and their cassandra data types from the dataframe
+    Args:
+        data_frame (DataFrame): the dataframe whose columns need to be extracted
+    Returns: dict
+    """
+    column_dtype = _get_data_frame_column_types(data_frame)
+    cassandra_columns = {key: _PYTHON_TO_CASSANDRA_DATA_TYPE_MAP[value] for (key, value) in
+                         column_dtype.items()}
+    return cassandra_columns
+
+
+def _pandas_factory(colnames, rows):
+    return pd.DataFrame(rows, columns=colnames)
+
+
+def _prepare_batches(prepared_statement, rows) -> list:
+    batches = []
+    itr = 0
+    batch = BatchStatement(consistency_level=ConsistencyLevel.QUORUM)
+    pbar = tqdm.tqdm(total=len(rows))
+    log.info("Preparing cassandra batches out of rows")
+    for row in rows:
```

Comment:
Would creating chunked dataframes help here? Rather than reading row by row? 

https://stackoverflow.com/questions/44729727/pandas-slice-large-dataframe-in-chunks/44729807

You can even drop to using the numpy interface to create chunks.

https://docs.scipy.org/doc/numpy/reference/generated/numpy.split.html

This approach might help you separate (and test) the functionality of chunking a dataframe and creating batch statement

Reply:
this is method is converting a list of tuples into a list of BatchStatement
there is no data frame involved.

this is also easier to extend into multithreading in future.

Reply:
updated to add more clarity